File: problems_list/0078.子集/0078-子集.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> ans;
        int n=nums.size();
        for(int i=0; i<pow(2, n); i++){
            vector<int> tmp;
            int idx=i;
            for(int j=0; j<n; j++){
                if(idx&1){
                    tmp.push_back(nums[j]);
                }
                idx>>=1;
            }
            ans.push_back(tmp);
        }
        return ans;
    }
};

File: problems_list/0078.子集/0078-子集.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsets = function(nums) {
    let ans=[];
    let n=nums.length;
    for(let i=0; i<Math.pow(2, n); i++){
        let curPerm=[];
        let idx=i;
        for(let j=0; j<n; j++){
            if(idx&1){
                curPerm.push(nums[j]);
            }
            idx>>=1;
        }
        ans.push([...curPerm]);
    }
    return ans;
};